examples/rust_caster.rs

```diff
@@ -264,16 +264,43 @@ fn play_media(
     }
 }
 
+
+fn discover() -> Option<(String, u16)> {
+    use mdns_sd::{ServiceDaemon, ServiceEvent};
+
+    let mdns = ServiceDaemon::new().expect("Failed to create daemon");
+
+    const SERVICE_TYPE: &str = "_googlecast._tcp.local.";
+    let receiver = mdns.browse(SERVICE_TYPE).expect("Failed to browse");
+
+    while let Ok(event) = receiver.recv() {
+        match event {
+            ServiceEvent::ServiceResolved(info) => {
+                return Some((info.get_addresses().iter().next().unwrap().clone().to_string(), info.get_port()))
+            }
+            _ => {}
+        }
+    }
+    return None
+}
```

Comment:
__note:__ how about about adding more debug info to this method? Crate examples are frequently used for debugging, so that should be valuable. Here's my proposal, but feel free to adjust it as you wish:
```rust
fn discover() -> Option<(String, u16)> {
    use mdns_sd::{ServiceDaemon, ServiceEvent};

    let mdns = ServiceDaemon::new().expect("Failed to create mDNS daemon.");

    const SERVICE_TYPE: &str = "_googlecast._tcp.local.";
    let receiver = mdns
        .browse(SERVICE_TYPE)
        .expect("Failed to browse mDNS services.");

    while let Ok(event) = receiver.recv() {
        match event {
            ServiceEvent::ServiceResolved(info) => {
                let mut addresses = info
                    .get_addresses()
                    .iter()
                    .map(|address| address.to_string())
                    .collect::<Vec<_>>();
                println!(
                    "{}{}",
                    Green.paint("Resolved a new service: "),
                    Red.paint(format!(
                        "{} ({})",
                        info.get_fullname(),
                        addresses.join(", ")
                    ))
                );

                // Based on mDNS crate code we should have at least one address available.
                return Some((addresses.remove(0), info.get_port()));
            }
            other_event => {
                println!(
                    "{}{}",
                    Green.paint("Received other service event: "),
                    Red.paint(format!("{:?}", other_event))
                );
            }
        }
    }

    None
}
```